cups/http-support.c

```diff
@@ -727,9 +727,9 @@ httpEncode64_2(char       *out,		/* I - String to write to */
     if (outptr < outend)
     {
       if (inlen > 1)
-        *outptr ++ = base64[(((in[0] & 255) << 4) | ((in[1] & 255) >> 4)) & 63];
+        *outptr ++ = base64[((in[0] << 4) | (in[1] >> 4)) & 63];
```

Comment:
"in" is sometimes signed char (implementation-specific), so we need to mask to 8 bits (sign extended when converted to int).

Reply:
Bitwise operations make that redundant. Like "& 63" already does the job and so does the << 4

Reply:
Without masking, the interior `in[1] >> 4` will get sign-extended so the upper 4 bits will be 1's. If "in" was `unsigned char` then it wouldn't be an issue, but "char" can be signed or unsigned.

Reply:
& 63 already does the necessary masking is what I am trying to say

Reply:
No, it doesn't.

`-1 >> 4` is still -1 (all 1's), which will mask the bits from the first part. `(-1 & 255) >> 4` is 15 (255 >> 4). So we need to mask `in[1]`...

Reply:
Okay, addressed